lib/takeaway.rb

```diff
@@ -0,0 +1,48 @@
+class Takeaway
+  attr_reader :basket
+
+  def initialize( restaurant, menu)
+    @menu = menu
+    @restaurant = restaurant
+    @basket = {}
+  end
+
+  def add( name, qty = 1)
+    fail "#{name} not available" unless @menu.contains?(name)
+    if @basket.has_key?(name) then
+      @basket[name] += qty
+    else
+      @basket[name] = qty
+    end
+  end
+
+  def remove ( name, qty = 1)
+    fail "#{name} not in basket" unless @basket.include?(name)
+    fail "Insuffient quantity of #{name} in basket" unless (basket[name] >= qty)
+    @basket[name] -= qty
+    @basket.delete(name) if @basket[name] == 0
+  end
+
+  def menu
+    puts @menu
+  end
+
+  def summary
+    str = ""
+    basket.each { |name, qty|
```

Comment:
Style/BlockDelimiters: Avoid using `{...}` for multi-line blocks. (https://github.com/bbatsov/ruby-style-guide#single-line-blocks)